File: .extra_files/setups/CallerId/SistemlerCom_CallerID_SDK/VisualStudio2010CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            axCIDv51.Hide();
            axCIDv51.Start();
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
            label1.Text = "Cihaz : " + axCIDv51.Command("Devicemodel") + "     " + axCIDv51.Command("Serial");
        }

        private void axCIDv51_OnCallerID(object sender, Axcidv5callerid.ICIDv5Events_OnCallerIDEvent e)
        {
         listBox1.Items.Add("Tel" + e.line + " - " + e.dateTime + " - " + e.phoneNumber);
            //  e.dateTime Telefon santrali tarafından gönderilen zaman bilgisidir. Saat ve dakika olarak gelir saniye gelmez.
            //  PC zamanından bağımsızdır. Gerekirse PC zamanı da kullanılabilir.
            //  Nesne .start  ile başlatılmışsa ve numara algılanmışsa, bu olay tetiklenir.

        }


    }
}

File: .extra_files/setups/CallerId/SistemlerCom_CallerID_SDK/VisualStudio2008VBNET/WindowsApplication1/Form1.vb
Public Class Form1

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        AxCIDv51.Hide()
        AxCIDv51.Start()



        ' Nesne başlatıldıktan sonra tekrar başlatılmaya çalışılırsa her hangi bir zararı veya faydası olmaz. Fakat anlamı da yoktur.
        ' Cihazın algılanması otomatiktir.
        ' Başka bir program cihazla bağlantı sağlamışsa, 
        ' Bu progrogram cihazla haberleşemez, numarayı gösteremez.
        ' Çalışan diğer program kapatılırsa otomatik bağlantı sağlanır.
        ' Program numarayı göstermeye başlar

    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        Dim Durum As String
        Durum = AxCIDv51.Command("devicemodel") 'cihaz modelini almaya yarar

        If Durum = "" Then
            Label1.Text = "Bağlantı yok"
        Else
            Label1.Text = "Cihaz Modeli : " + Durum + " Seri no : " + AxCIDv51.Command("serial")
        End If
        
    End Sub

    Private Sub AxCIDv51_OnCallerID(ByVal sender As System.Object, ByVal e As Axcidv5callerid.ICIDv5Events_OnCallerIDEvent) Handles AxCIDv51.OnCallerID
        Label2.Text = e.phoneNumber

        TextBox1.Text = " Hat no: " + e.line + " - Zaman: " + Now + " Arayan: " + e.phoneNumber + Chr(13) + Chr(10) + TextBox1.Text
        'TextBox1.Text = " Hat no: " + e.line + " - Zaman: " + e.dateTime + " Arayan: " + e.phoneNumber + Chr(13) + Chr(10) + TextBox1.Text
        'e.dateTime Telefon santrali tarafından gönderilen zaman bilgisidir. 
        'PC zamanından bağımsızdır. Gerekirse kullanılabilir.
        'Nesne .start  ile başlatılmışsa ve numara algılanmışsa, bu olay tetiklenir.

    End Sub

    Private Sub AxCIDv51_OnSignalA(ByVal sender As System.Object, ByVal e As Axcidv5callerid.ICIDv5Events_OnSignalAEvent) Handles AxCIDv51.OnSignalA
        ProgressBar1.Value = e.level1
        ProgressBar2.Value = e.level2
        ProgressBar3.Value = e.level3
        ProgressBar4.Value = e.level4
        ' Bu göstergeleri kullanmak isteğe bağlıdır. Kablo bağlantısının doğru olup olmadığını anlamaya yarar. Test amaçlıdır. 
        ' Bu olay periyodik olarak tetiklenmektedir.
        ' level4 den sonrakiler anlamlı değildir.
        ' Çünkü  V5 serisi cihazlar 4 hatlı üretilmektedir. 8 hatlı v5 cihaz yoktur.

    End Sub
End Class
